### PR TITLE
Add unit tests for PR "Cancel web authorization session if application is not active"

### PR DESCRIPTION
**Pull request by aleksandr-dorofeev**
Things to consider before you submit the PR:

* ~[ ] Has `CHANGELOG.md` been updated?~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* ~[ ] Did you check UI tests on the sample app? They are not executed on CI.~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR contains unit tests for PR [Cancel web authorization session if application is not active](https://github.com/microsoft/appcenter-sdk-apple/pull/2374).

## Related PRs or issues

[AB89320](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/89320)

### CODE REVIEW

File: AppCenterDistribute/AppCenterDistributeTests/MSACDistributeTests.m [hunk 1]
```diff
@@ -1769,6 +1769,82 @@ - (void)testBrowserNotOpenedWhenTesterAppUsedForUpdateSetup {
...
+  OCMStub(ClassMethod([distributeUtilMock getPresentationAnchor])).andReturn(nil);
+  OCMReject([distributeMock openURLInAuthenticationSessionWith:OCMOCK_ANY usePresentationContext:OCMOCK_ANY]);
+  OCMStub([distributeMock openURLInAuthenticationSessionWith:OCMOCK_ANY usePresentationContext:OCMOCK_ANY]).andDo(nil);
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Web authorization canceled"];
+
+  // When
+  [MSAC_APP_CENTER_USER_DEFAULTS setObject:@"requestId" forKey:kMSACUpdateTokenRequestIdKey];
+  [self.sut setUpdateFlowInProgress:YES];
+  [self.sut openUrlInAuthenticationSessionOrSafari:[NSURL URLWithString:@"some_url://"]];
+
```
Comment (DmitriyKirakosyan) [inline AppCenterDistribute/AppCenterDistributeTests/MSACDistributeTests.m:1786]:
```suggestion
```

File: AppCenterDistribute/AppCenterDistributeTests/MSACDistributeTests.m [hunk 2]
```diff
@@ -1769,6 +1769,82 @@ - (void)testBrowserNotOpenedWhenTesterAppUsedForUpdateSetup {
...
+  id presentationAnchorMock = OCMClassMock([UIWindow class]);
+  OCMStub(ClassMethod([distributeUtilMock getPresentationAnchor])).andReturn(presentationAnchorMock);
+  OCMStub([distributeMock openURLInAuthenticationSessionWith:OCMOCK_ANY usePresentationContext:OCMOCK_ANY]).andDo(nil);
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Web authorization opened"];
+
+  // When
+  [MSAC_APP_CENTER_USER_DEFAULTS setObject:@"requestId" forKey:kMSACUpdateTokenRequestIdKey];
+  [self.sut setUpdateFlowInProgress:YES];
+  [self.sut openUrlInAuthenticationSessionOrSafari:[NSURL URLWithString:@"some_url://"]];
+
```
Comment (DmitriyKirakosyan) [inline AppCenterDistribute/AppCenterDistributeTests/MSACDistributeTests.m:1822]:
```suggestion
```

**Review by DmitriyKirakosyan — CHANGES_REQUESTED** [review]

File: AppCenterDistribute/AppCenterDistributeTests/MSACDistributeTests.m [hunk 3]
```diff
@@ -1769,6 +1769,82 @@ - (void)testBrowserNotOpenedWhenTesterAppUsedForUpdateSetup {
...
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [expectation fulfill];
+  });
+
+  // Then
+  [self waitForExpectationsWithTimeout:1
+                               handler:^(NSError *error) {
+                                 // Then
```
Comment (AnastasiaKubova) [inline AppCenterDistribute/AppCenterDistributeTests/MSACDistributeTests.m:1794]: empty line :)

File: AppCenterDistribute/AppCenterDistributeTests/MSACDistributeTests.m [hunk 4]
```diff
@@ -1769,6 +1769,82 @@ - (void)testBrowserNotOpenedWhenTesterAppUsedForUpdateSetup {
...
```
Comment (AnastasiaKubova) [inline AppCenterDistribute/AppCenterDistributeTests/MSACDistributeTests.m:1830]: and here

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**